documents/invoice/model.go

```diff
@@ -427,3 +427,32 @@ func (i *Invoice) CreateNFTProofs(
 		i.DocumentType(),
 		account, registry, tokenID, nftUniqueProof, readAccessProof)
 }
+
+// CollaboratorCanUpdate checks if the collaborator can update the document.
+func (i *Invoice) CollaboratorCanUpdate(updated documents.Model, did identity.DID) error {
```

Comment:
just a naming comment, but maybe "did" can be adjusted to reflect the method name --> collaborator